pkg/apis/eksctl.io/v1alpha5/validation_test.go

```diff
@@ -469,6 +472,29 @@ var _ = Describe("ClusterConfig validation", func() {
 				err = validateInstancesDistribution(ng)
 				Expect(err).ToNot(HaveOccurred())
 			})
+
+			It("It fails when the spotAllocationStrategy is not a supported strategy", func() {
+				ng.InstancesDistribution.SpotAllocationStrategy = &testUnsupportedSpotAllocationStrategy
```

Comment:
don't you need `ng.InstancesDistribution.SpotInstancePools = newInt(2)` here too?

Reply:
it would actually check whether the allocation strategy is defined with or without spotInstancePools. I'd like to leave it out though in case the validation logic would ever get changed to check if spotInstancePools is specified and spotAllocationStrategy is not "lowest-price" (since that's the only allocation strategy where it is valid to specify spotInstancePools).

Reply:
Good point :+1: